ktor-io/common/src/io/ktor/utils/io/ByteChannel.kt

```diff
@@ -70,7 +70,7 @@ public class ByteChannel(public val autoFlush: Boolean = false) : ByteReadChanne
         }
 
         if (_readBuffer.size < CHANNEL_MAX_SIZE) moveFlushToReadBuffer()
-        return _closedCause.value == null
+        return _readBuffer.size >= min
```

Comment:
@e5l I noticed this small issue in `ByteChannel` when testing the `skipIfFound` function - after the writer closes the channel, then the reader would always return false even if the required content is available.